docker_services_cli/services.py

```diff
@@ -8,30 +8,147 @@
+            stderr=subprocess.DEVNULL
+        )
+        return True
+    except subprocess.CalledProcessError:
+        return False
+
+
+def es_healthcheck(*args, **kwargs):
+    """Elasticsearch healthcheck."""
+    return _run_healthcheck_command([
+        "curl",
+        "-f",
+        "localhost:9200/_cluster/health?wait_for_status=green"
+    ])
+
+
+def postgresql_healthcheck(*args, **kwargs):
+    """Postgresql healthcheck."""
+    filepath = kwargs['filepath']
+
+    return _run_healthcheck_command([
+        "docker-compose",
+        "--file",
+        filepath,
+        "exec",
+        "postgresql",
+        "bash",
+        "-c",
+        "pg_isready",
+    ])
+
+
+def mysql_healthcheck(*args, **kwargs):
+    """Mysql healthcheck."""
+    filepath = kwargs['filepath']
+    password = MYSQL["MYSQL_ROOT_PASSWORD"]
+
+    return _run_healthcheck_command([
+        "docker-compose",
+        "--file",
+        filepath,
+        "exec",
+        "mysql",
+        "bash",
+        "-c",
+        f"mysql -p{password} -e \"select Version();\"",
+    ])
+
+
+def redis_healthcheck(*args, **kwargs):
+    """Redis healthcheck."""
+    filepath = kwargs['filepath']
+
+    return _run_healthcheck_command([
+        "docker-compose",
+        "--file",
+        filepath,
+        "exec",
+        "redis",
+        "bash",
+        "-c",
+        "redis-cli ping",
+        "|",
+        "grep 'PONG'",
+        "&>/dev/null;",
+    ])
+
+
+READYNESS_CHECKS = {
+    "es": es_healthcheck,
+    "postgresql": postgresql_healthcheck,
+    "mysql": mysql_healthcheck,
+    "redis": redis_healthcheck,
+}
+"""Readyness check functions module path, as string."""
+
+
+def wait_for_services(services, filepath=DOCKER_SERVICES_FILEPATH, max_retries=6):
+    """Wait for services to be up.
+
+    It performs configured healthchecks in a serial fashion, following the
+    order given in the ``up`` command.
     """
-    command = ["docker-compose", "--file", filepath, "down"]
+    for service in services:
+        exp_backoff_time = 2
+        try_ = 1
+        # Using plain __import__ to avoid depending on invenio-base
+        check = READYNESS_CHECKS[service]
+        ready = check(filepath=filepath)
+        while not ready and try_ < max_retries:
+            click.secho(
+                f"{service} not ready at {try_} retries, waiting " \
+                f"{exp_backoff_time}s",
+                fg="yellow"
+            )
+            try_ += 1
+            time.sleep(exp_backoff_time)
+            exp_backoff_time *= 2
+            ready = check(filepath=filepath)
 
-    subprocess.run(command, check=True)
+        if not ready:
+            click.secho(f"Unable to boot up {service}", fg="red")
+            exit(1)
+        else:
+            click.secho(f"{service} up and running!", fg="green")
 
 
-def services_up(services, filepath=DOCKER_SERVICES_FILEPATH):
+def services_up(services, filepath=DOCKER_SERVICES_FILEPATH, wait=True):
```

Comment:
`services` needs a default (which should be all services). If one does:
```console
$ docker-services-cli up
```

The `wait_for_services` doesn't do anything because `services` is an empty list.